src/scenic/syntax/veneer.py

```diff
@@ -111,1000 +132,1341 @@
+        modifiable_props={'position'})
+
+@distributionFunction
+def findOnHelper(region, pos, on_direction):
+    on_pos = region.findOn(pos, on_direction=on_direction)
 
-	Specifies :prop:`position`, with no dependencies. Optionally specifies
-	:prop:`heading` if the given `Region` has a :term:`preferred orientation`.
-	"""
-	region = toType(region, Region, 'specifier "in/on R" with R not a Region')
-	extras = {'heading'} if alwaysProvidesOrientation(region) else {}
-	return Specifier('position', Region.uniformPointIn(region), optionals=extras)
+    if on_pos is None:
+        raise RejectionException("Unable to place object on surface.")
+    else:
+        return on_pos
 
 def alwaysProvidesOrientation(region):
-	"""Whether a Region or distribution over Regions always provides an orientation."""
-	if isinstance(region, Region):
-		return region.orientation is not None
-	elif (isinstance(region, Options)
-		  and all(alwaysProvidesOrientation(opt) for opt in region.options)):
-		return True
-	else:	# TODO improve somehow!
-		try:
-			sample = region.sample()
-			return sample.orientation is not None or sample is nowhere
-		except RejectionException:
-			return False
+    """Whether a Region or distribution over Regions always provides an orientation."""
+    if isinstance(region, Region):
+        return region.orientation is not None
+    elif (isinstance(region, MultiplexerDistribution)
+          and all(alwaysProvidesOrientation(opt) for opt in region.options)):
+        return True
+    else:   # TODO improve somehow!
+        try:
+            sample = region.sample()
+            return sample.orientation is not None or sample is nowhere
+        except RejectionException:
+            return False
 
-def Beyond(pos, offset, fromPt=None):
-	"""The :specifier:`beyond {X} by {Y} from {Z}` polymorphic specifier.
-
-	Specifies :prop:`position`, with no dependencies.
-
-	Allowed forms::
-
-		beyond <vector> by <number> [from <vector>]
-		beyond <vector> by <vector> [from <vector>]
-
-	If the :grammar:`from <vector>` is omitted, the position of ego is used.
-	"""
-	pos = toVector(pos, 'specifier "beyond X by Y" with X not a vector')
-	offset = toTypes(offset, (Vector, builtins.float),
-	                 'specifier "beyond X by Y" with Y not a number or vector')
-	dType = underlyingType(offset)
-	if dType is builtins.float:
-		offset = Vector(0, offset)
-	elif dType is not Vector:
-		raise RuntimeParseError('specifier "beyond X by Y" with Y not a number or vector')
-	if fromPt is None:
-		fromPt = ego()
-	fromPt = toVector(fromPt, 'specifier "beyond X by Y from Z" with Z not a vector')
-	lineOfSight = fromPt.angleTo(pos)
-	return Specifier('position', pos.offsetRotated(lineOfSight, offset))
+def OffsetBy(offset):
+    """The :grammar:`offset by <vector>` specifier.
 
-def VisibleFrom(base):
-	"""The :grammar:`visible from <Point>` specifier.
+    Specifies :prop:`position` and :prop:`parentOrientation`, with no dependencies.
```

Comment:
This docstring and the next several should say that `parentOrientation` is only specified optionally (in fact it would probably be good to give the priorities in these docstrings when not equal to 1).